src/Adapter/MSTest.TestAdapter/Discovery/AssemblyEnumerator.cs

```diff
@@ -265,7 +265,94 @@ private bool DynamicDataAttached(IDictionary<string, object?> sourceLevelParamet
         TestMethod testMethod = test.TestMethod;
         MSTestAdapter.PlatformServices.Interface.ITestContext testContext = PlatformServiceProvider.Instance.GetTestContext(testMethod, writer, sourceLevelParameters);
         TestMethodInfo? testMethodInfo = _typeCache.GetTestMethodInfo(testMethod, testContext, MSTestSettings.CurrentSettings.CaptureDebugTraces);
-        return testMethodInfo != null && TryProcessTestDataSourceTests(test, testMethodInfo, tests);
+
+        if (testMethodInfo == null)
+        {
+            return false;
+        }
+
+        // Add non-runnable tests like AssemblyInitialize, AssemblyCleanup, ClassInitialize, ClassCleanup.
+        if (MSTestSettings.CurrentSettings.FixturesEnabled)
+        {
+            AddNonRunnableTests(testMethodInfo, tests, nonRunnableTests);
+        }
```

Comment:
I think this deserves to be called directly as it's not linked to dynamic data feature.

Reply:
I need testMethodInfo object. That's I am populating here. If we can get this object in a higher level, we can poopulate over there as well

Reply:
updated,